### PR TITLE
[Fixes #2095] Mark IControllerFactory methods in DefaultControllerFactor...

### LINKED ISSUE #2095: Mark IControllerFactory methods in DefaultControllerFactory as virtual
Methods I'm referring to are CreateController and ReleaseController. These extension points are sorely needed.

### PR DESCRIPTION
**Pull request by kichalla**
...y as virtual

@rynowak

### CODE REVIEW

**Comment by Eilon** [comment]
:shipit:

**Comment by rynowak** [comment]
:shipit:

File: src/Microsoft.AspNet.Mvc.Core/DefaultControllerFactory.cs [hunk 1]
```diff
@@ -38,8 +38,19 @@ public DefaultControllerFactory(IControllerActivator controllerActivator)
             _getPropertiesToActivate = GetPropertiesToActivate;
         }
 
+        /// <summary>
+        /// The <see cref="IControllerActivator"/> used to create a controller.
+        /// </summary>
+        protected IControllerActivator ControllerActivator
```
Comment (harshgMSFT) [inline src/Microsoft.AspNet.Mvc.Core/DefaultControllerFactory.cs]: I am wondering ControllerActivator is a transient service, while ControllerFactory is a singleton are we not increasing the lifetime  of the activator?

Comment (pranavkm) [inline src/Microsoft.AspNet.Mvc.Core/DefaultControllerFactory.cs]: The factory caches things, consequently it's a singleton. Everything consumed by a singleton ends up having a singleton lifetime.

Comment (harshgMSFT) [inline src/Microsoft.AspNet.Mvc.Core/DefaultControllerFactory.cs]: Yea, unfortunately we can't do anything about this. Oh well.

Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Core/DefaultControllerFactory.cs]: I don't get the arguments why this is bad. Transient is the 'caller-decides' policy. We're increasing the lifetime of the activator passed to this class in the constructor by DI - but not any other instances, let's say requested by user code.

If the activator **owns** a cache I think there's an argument to be made for making it a singleton.

TLDR: Picked most flexible lifetime, used it in the way we want.
